wrappers/python/nntile/layer/t5_attention.py

```diff
@@ -0,0 +1,1580 @@
+        if self.in_proj_bias_k is not None:
+            add_fiber_inplace_async(1, self.in_proj_bias_k.value, 1, k_partial, 0, 1)
+
+        return k_partial
+
+    def _forward_mlp_v_async(self):
+        # V_transposed = einsum('jkl,lmn->jkmn', W_V, X_V)
+        # gemm (n_head, head_size, n_emb) by (n_emb, n_seq, n_batch) into
+        # (n_head, head_size, n_seq, n_batch)
+        gemm_async(
+            1.0,
+            notrans,
+            self.w_v.value,
+            notrans,
+            self.x_v.value,
+            0.0,
+            self.v_transposed.value,
+            1,
+            0,
+            redux=self.redux,
+        )
+        # Rotate axes into (head_size, n_seq, n_batch, n_head)
+        transpose_async(1.0, self.v_transposed.value, self.v.value, 1)
+        # X_V, W_V and V_transposed can be offloaded from GPU
+        self.x_v.value.wont_use()
+        # self.v_transposed.value.wont_use()
+        self.v_transposed.value.invalidate_submit()
+        self.w_v.value.wont_use()
+        # Apply bias if needed
+        if self.in_proj_bias_v is not None:
+            # batched add_fiber_inplace (head_size, batch=n_head) into
+            # (head_size, n_seq, n_batch, batch=n_head)
+            add_fiber_inplace_async(1, self.in_proj_bias_v.value, 1, self.v.value, 0, 1)
+            self.in_proj_bias_v.value.wont_use()
+
+    def _forward_mlp_v_dynamic(self, x: Tensor):
+        v_partial_tr = self._get_tmp_tr_for_cache(x)
+        v_partial = self._get_tmp_for_cache(x)
+
+        gemm_async(
+            1.0,
+            notrans,
+            self.w_v.value,
+            notrans,
+            x,
+            0.0,
+            v_partial_tr,
+            1,
+            0,
+            redux=self.redux,
+        )
+
+        transpose_async(1.0, v_partial_tr, v_partial, 1)
+
+        if self.in_proj_bias_v is not None:
+            add_fiber_inplace_async(1, self.in_proj_bias_v.value, 1, v_partial, 0, 1)
+
+        return v_partial
+
+    def _add_positional_bias_async(self):
+        query_length, key_length = self.a.value.shape[:2]
+        context_position = np.arange(query_length, dtype=np.int32)[:, None]
+        memory_position = np.arange(key_length, dtype=np.int32)[None, :]
+        relative_position = (
+            memory_position - context_position
+        )  # shape (query_length, key_length)
+        relative_position_bucket_np_value = relative_position_bucket_numpy(
+            relative_position,
+            bidirectional=(not self.is_decoder),
+            num_buckets=32,
+            max_distance=128,
+        )
+        relative_position_bucket_np_value = relative_position_bucket_np_value[None, ...]
+        if self.a.value.shape[2] > 1:
+            relative_position_bucket_np_value = (
+                relative_position_bucket_np_value.repeat(self.a.value.shape[2], axis=0)
+            )
+
+        self.relative_position_bucket_nnt = nntc.from_array(
+            relative_position_bucket_np_value.T,
```

Comment:
This tensor is of type int32, but embedding layer requires int64. There is no type Tensor_int32,  but input type np.int32 is mapped into Tensor_int64. There might some bugs, related to such a conversion.